Lab1 submission

guscarrian opened the pull request:
Movie ticket booking system - It's still a work in progress!

guscarrian committed:
adding key

guscarrian committed:
work in progress

guscarrian committed:
Merge branch 'vladmaraev:master' into lab1

guscarrian committed:
progress but issue with context, damn it

vladmaraev commented:
It is not a final submission, right? If not please feel free to write your question here.

guscarrian commented:
I'm having issues with the context. 
Originally my grammar looked like this:

const grammar = {
    entities: {
      movie: "titanic",
      date: "friday",
      showtime: "7:00",
    },
  }

I have three slots: the movie name (movie), the date when the user wants to go to the cinema (date), and the time when the movie is played (showtime). 

I created a function (checkWordExists) to convert the user's input utterance to lowecase, remove any trailing period and split it into an array of words (defined in line 53).

A example interaction could be:
S: Hi there! How can I help you?
U: I want to watch a movie.
S: What movie do you want to watch?
U: I'd like to watch Titanic.
S: When do you want to go to the cinema?
U: On Friday.
S: What time do you want to watch (context.Movie) on (context.Date)?
U: At 7:00
S: Sure! I will book a ticket for (context.Movie) on (context.Date) at (context.Showtime)

The user could also say "I'd like to watch Titanic on Friday" or "I want to watch Titanic on Friday at 7:00" in one shot. However, since the grammar was so limited, I decided to try:

const grammar = {
    entities: {
    movie: ["barbie", "titanic", "avatar"],
    date: ["tonight", "tomorrow", "friday"],
    showtime: ["7:00", "7 pm", "10:00"],
    },
  };

The checkWordExists function was also updated (line 130) to lowercase, remove trailing periods and split the user's utterance, and to check if any of the words in the user's input is in the list of the chosen entity. 

The issue with the context is that the system now returns all items in each entity list. For example, instead of:
S: What time do you want to watch (Titanic) on (Friday)?
It now returns:
S: What time do you want to watch (Barbie, Titanic, Avatar) on (tonight, tomorrow, Friday)?

guscarrian committed:
testing

vladmaraev commented:
I hope I understood your issue correctly. I would suggest having a grammar that would accept realistic utterances:
```js
const grammarAll = {
"I'd like to watch Titanic on Friday": 
       {entities: {movie: "titanic",
                        date: "friday"}},
// etc
}
```
as well as per-slot grammars:
```js
const grammarDate = {"friday": {entities: {date: "friday"}}, 
// etc
}
```

You can also join these grammars together if you expect both inputs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals
